ios/instruments/instruments_sysmontap.go

```diff
@@ -0,0 +1,114 @@
+package instruments
+
+import (
+	"fmt"
+
+	"github.com/danielpaulus/go-ios/ios"
+	dtx "github.com/danielpaulus/go-ios/ios/dtx_codec"
+)
+
+const sysmontapName = "com.apple.instruments.server.services.sysmontap"
+
+type SysmontapService struct {
+	channel    *dtx.Channel
+	conn       *dtx.Connection
+	plistCodec ios.PlistCodec
+}
+
+// Creates a new SysmontapService
+func NewSysmontapService(device ios.DeviceEntry) (*SysmontapService, error) {
+	dtxConn, err := connectInstruments(device)
+	if err != nil {
+		return nil, err
+	}
+	processControlChannel := dtxConn.RequestChannelIdentifier(sysmontapName, loggingDispatcher{dtxConn})
+
+	return &SysmontapService{channel: processControlChannel, conn: dtxConn, plistCodec: ios.NewPlistCodec()}, nil
+}
+
+// Close closes up the DTX connection
+func (s *SysmontapService) Close() {
+	s.conn.Close()
+}
+
+// Start sends a start method call async and waits until the cpu info & stats come back
+func (s *SysmontapService) Start() (SysmontapMessage, error) {
+	err := s.channel.MethodCallAsync("start")
+	if err != nil {
+		return SysmontapMessage{}, err
+	}
+
+	globalChannel := s.conn.GlobalChannel()
+
+	// Receive() will block until the message with the CPU usage is delivered
+	msg, err := globalChannel.Receive()
+	if err != nil {
+		return SysmontapMessage{}, err
+	}
+
+	sysmontapMessage, err := mapToCPUUsage(msg)
+	if err != nil {
+		return SysmontapMessage{}, err
+	}
+
+	return sysmontapMessage, nil
+}
+
+// SetConfig sets configuration to allow the sysmontap service getting desired data points
+func (s *SysmontapService) SetConfig(procAttrs, sysAttrs []interface{}) error {
+	config := map[string]interface{}{
+		"ur":             500,
+		"bm":             0,
+		"procAttrs":      procAttrs,
+		"sysAttrs":       sysAttrs,
+		"cpuUsage":       true,
+		"physFootprint":  true,
+		"sampleInterval": 500000000,
+	}
```

Comment:
I see that in go-ios we already use DTServiceInfoDevice, I checked what can come from there https://github.com/search?q=repo%3AMTACS%2FiOS-17-Runtime-Headers%20sysmonProcessAttributes&type=code 

And in this header it looks like we expect these sysAttr and procAttr: https://github.com/MTACS/iOS-17-Runtime-Headers/blob/d1d960dfaa4107765dd7fcf891e4967c0930d5fd/PrivateFrameworks/DVTInstrumentsFoundation.framework/DTSysmonTapConfig.h#L9

`setConfig` initially was found here: https://github.com/MTACS/iOS-17-Runtime-Headers/blob/d1d960dfaa4107765dd7fcf891e4967c0930d5fd/protocols/DTTapAuthorizedAPI.h#L8 

it looks like a generic protocol for launching any metrics of the device

Based on your discovery, I see that it is related to this https://github.com/MTACS/iOS-17-Runtime-Headers/blob/d1d960dfaa4107765dd7fcf891e4967c0930d5fd/PrivateFrameworks/DVTInstrumentsFoundation.framework/DTActivityTraceTapConfig.h#L19 , but I don't see if there is any way to get the CPU data and looks like it is more oriented on network traffic, but a good find - we could use also later

Reply:
I can't really see anything about the values in the config in those links. The main point was that changing `ur: 500` to `ur: 10` improves the frequency of which samples arrive drastically. Instead of waiting 10+s for the first sample to arrive you get it basically instantly.